modules/swagger-compat-spec-parser/src/main/java/io/swagger/parser/SwaggerCompatConverter.java

```diff
@@ -186,7 +191,22 @@ public ResourceListing readResourceListing(String input, MessageBuilder messages
                 String json = RemoteUrl.urlToString(input, auths);
                 jsonNode = Json.mapper().readTree(json);
             } else {
-                jsonNode = Json.mapper().readTree(new File(input));
+                final String fileScheme = "file:";
+                java.nio.file.Path path;
+                if (input.toLowerCase().startsWith(fileScheme)) {
```

Comment:
To avoid NPE, to compare with fileSchema, as it is constant

Reply:
No, it can't throw NPE on line 196, it'll throw it in the previous if(input.startsWith("http")) statement.